WebPhapp/backend/server.js

```diff
@@ -215,54 +109,159 @@ app.post('/api/v1/prescriptions/add',(req,res) => {
     // TODO: query blockchain to get current highest prescriptionID
     prescription.prescriptionID = -1;
 
-    // Add the prescription to the blockchain and index this prescription.
+    // Add the prescription to the blockchain and index this prescription in MySQL.
     return finish("TODO: build prescription add to blockchain", true);
 });
 
-// An api endpoint that returns the prescription associated with a
-// given prescription ID.
-// example: http://localhost:5000/api/v1/prescriptions/single/0002
-app.get('/api/v1/prescriptions/single/:prescriptionID', (req,res) => {
-    var prescriptionID = parseInt(req.params.prescriptionID);
+/*
+An api endpoint that returns all of the prescriptions associated with a patient ID
+Examples:
+    Directly in terminal:
+        >>> curl "http://localhost:5000/api/v1/prescriptions/1"
+    To be used in Axois call:
+        .get("api/v1/prescriptions/1")
+Returns:
+    A list of prescription objects each with fields: [
+        prescriptionID, patientID, drugID, fillDates,
+        writtenDate, quantity, daysFor, refillsLeft,
+        prescriberID, dispenserID, cancelDate, drugName
+    ]
+*/
+app.get('/api/v1/prescriptions/:patientID', (req,res) => {
+    var patientID = parseInt(req.params.patientID);
     var prescriptions = readJsonFileSync(
         __dirname + '/' + "dummy_data/prescriptions.json").prescriptions;
 
-    var prescription = prescriptions.find( function(elem) {
-        return elem.prescriptionID === prescriptionID;
+    var toSend = [];
+    prescriptions.forEach(prescription => {
+        if (prescription.patientID === patientID) toSend.push(prescription);
```

Comment:
Not a huge fan of this condensed if notation. Makes the readability go down, in my opinion.